lib/ts/types.ts

```diff
@@ -17,25 +17,26 @@ import RecipeModule from "./recipeModule";
 import NormalisedURLDomain from "./normalisedURLDomain";
 import NormalisedURLPath from "./normalisedURLPath";
 import { TypeFramework } from "./framework/types";
+import { BaseRequest } from "./framework";
 
 export type AppInfo = {
     appName: string;
-    websiteDomain: string;
-    websiteBasePath?: string;
+    origin: string | ((req: BaseRequest, userContext: any) => Promise<string | undefined>);
```

Comment:
```suggestion
    origin: string | ((req: BaseRequest, userContext: any) => Promise<string>);
```

Reply:
We remove the undefined cause for requests from dashboard, we still  force users to return a domain so that it works for them.